### PR TITLE
[SPARK-47597][STREAMING] Streaming: Migrate logInfo with variables to structured logging framework

### PR DESCRIPTION
**Pull request by dtenedor**
### What changes were proposed in this pull request?

Migrate logInfo with variables of the streaming module to structured logging framework. This transforms the logInfo entries of the following API
```
def logInfo(msg: => String): Unit
```
to
```
def logInfo(entry: LogEntry): Unit
```

### Why are the changes needed?

To enhance Apache Spark's logging system by implementing structured logging.

### Does this PR introduce _any_ user-facing change?

Yes, Spark core logs will contain additional MDC

### How was this patch tested?

Compiler and scala style checks, as well as code review.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by dtenedor** [comment]
@gengliangwang FYI

**Comment by dtenedor** [comment]
Note that the single test failure is flaky and unrelated; the latest CI run otherwise passes all tests.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala [hunk 1]
```diff
@@ -102,7 +102,7 @@ class IncrementalExecution(
       tracker).transformAllExpressionsWithPruning(
       _.containsAnyPattern(CURRENT_LIKE, EXPRESSION_WITH_RANDOM_SEED)) {
       case ts @ CurrentBatchTimestamp(timestamp, _, _) =>
-        logInfo(s"Current batch timestamp = $timestamp")
+        logInfo(log"Current batch timestamp = ${MDC(TIME_UNITS, timestamp)}")
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala:105]: It seems that we have been using `TIME_UNITS` for durations. This one is timestamp.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala [hunk 2]
```diff
@@ -171,7 +173,9 @@ class MicroBatchExecution(
             // Materialize source to avoid creating it in every batch
             val metadataPath = s"$resolvedCheckpointRoot/sources/$nextSourceId"
             nextSourceId += 1
-            logInfo(s"Reading table [$table] from DataSourceV2 named '$srcName' $dsStr")
+            logInfo(log"Reading table [${MDC(STREAMING_TABLE, table)}] " +
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala:176]: I wonder if we need the STREAMING prefix for these variables. But we can always revisit and combine variables later.

**Review by dtenedor — COMMENTED** [review]
Thanks @gengliangwang for your review!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala [hunk 3]
```diff
@@ -104,8 +105,8 @@ class ProgressReporter(
       addNewProgress(newProgress)
       if (lastNoExecutionProgressEventTime > Long.MinValue) {
         postEvent(new QueryIdleEvent(id, runId, formatTimestamp(currentTriggerStartTimestamp)))
-        logInfo(s"Streaming query has been idle and waiting for new data more than " +
-          s"${noDataProgressEventInterval} ms.")
+        logInfo(log"Streaming query has been idle and waiting for new data more than " +
+          log"${MDC(STREAMING_QUERY_PROGRESS, noDataProgressEventInterval)} ms.")
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala:109]: This one doesn't seem right

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala [hunk 4]
```diff
@@ -307,7 +311,8 @@ class ContinuousExecution(
     } catch {
       case t: Throwable if StreamExecution.isInterruptionException(t, sparkSession.sparkContext) &&
           state.get() == RECONFIGURING =>
-        logInfo(s"Query $id ignoring exception from reconfiguring: $t")
+        logInfo(log"Query ${MDC(QUERY_ID, id)} ignoring exception from reconfiguring: " +
+          log"${MDC(EXCEPTION, t)}")
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala:315]: Usually we use `ERROR` for this one.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala [hunk 5]
```diff
@@ -469,9 +475,10 @@ class RocksDBFileManager(
             log"version ${MDC(FILE_VERSION, version)}", e)
       }
     }
-    logInfo(s"Deleted ${filesToDelete.size - failedToDelete} files (failed to delete" +
-      s"$failedToDelete files) not used in versions >= $minVersionToRetain")
-
+    logInfo(log"Deleted ${MDC(NUM_FILES, filesToDelete.size - failedToDelete)} files " +
+      log"(failed to delete" +
+      log"${MDC(NUM_FILES_FAILED_TO_DELETE, failedToDelete)} files) " +
+      log"not used in versions >= ${MDC(VERSION_NUMBER, minVersionToRetain)}")
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBFileManager.scala:481]: MIN_VERSION_NUMBER?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala (hunk 4 above)
Comment (dtenedor) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala]: Nice, I missed that we already had that log key, switched everything to use that instead.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala (hunk 2 above)
Comment (WweiL) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/MicroBatchExecution.scala:176]: Also for streaming we have streaming query id which is useful in identifying which log is from which query. This is particularly useful when there are multiple streaming queries. Was wondering if this is worth added.

**Pull request closed without merging**